crates/bevy_asset/src/server/info.rs

```diff
@@ -189,7 +189,17 @@ impl AssetInfos {
         loading_mode: HandleLoadingMode,
         meta_transform: Option<MetaTransform>,
     ) -> Result<(UntypedHandle, bool), GetOrCreateHandleInternalError> {
-        match self.path_to_id.entry(path.clone()) {
+        // TODO: It is not currently possible to create or get a `Handle` without the `type_id` parameter.
+        // This function should
+        let type_id =
```

Comment:
Can you explain why you moved this check out here, above the `weak_handle.upgrade()` branch? Is there a plan for resolving this?

Reply:
The check had to be moved as now multiple handles could be stored against a single path, whereas before a path (including label) was a unique address for an asset. As such, the `TypeId` is required to disambiguate which asset is desired from a given path.

To resolve this TODO, I've added an additional check to see if only 1 asset is currently registered against the path, and if so, work with that as the assumed correct one. Since the previous behaviour was to only ever have 1 or 0 assets per path, this should preserve the existing behaviour.